projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -6469,19 +6473,29 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
      */
     endRowTransaction(commit: boolean, row: IgxRow) {
         row.newData = this.transactions.getAggregatedValue(row.id, true);
-
-        let args = row.createEditEventArgs();
+        const rowArgs = row.createEditEventArgs();
+        let rowEditArgs: IGridEditEventArgs;
 
         if (!commit) {
-            this.onRowEditCancel.emit(args);
+            this.rowEditExit.emit(rowArgs);
+            if (rowArgs.cancel) {
+                return true;
+            }
+
             this.transactions.endPending(false);
         } else {
-            args = this.gridAPI.update_row(row, row.newData);
-        }
-        if (args.cancel) {
-            this.transactions.startPending();
```

Comment:
I believe this pending was there for a reason (because update_row will clear the pending state). If this is not applied, changes after a cancel end up directly committed to the data (or as separate transactions) which is not intended.